dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/registry/RegistryCenter.java

```diff
@@ -0,0 +1,257 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.service.registry;
+
+import static org.apache.dolphinscheduler.common.Constants.REGISTRY_DOLPHINSCHEDULER_DEAD_SERVERS;
+
+import org.apache.dolphinscheduler.common.IStoppable;
+import org.apache.dolphinscheduler.common.utils.PropertyUtils;
+import org.apache.dolphinscheduler.common.utils.StringUtils;
+import org.apache.dolphinscheduler.spi.plugin.DolphinPluginLoader;
+import org.apache.dolphinscheduler.spi.plugin.DolphinPluginManagerConfig;
+import org.apache.dolphinscheduler.spi.register.Registry;
+import org.apache.dolphinscheduler.spi.register.RegistryException;
+import org.apache.dolphinscheduler.spi.register.RegistryPluginManager;
+import org.apache.dolphinscheduler.spi.register.SubscribeListener;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableList;
+
+/**
+ * All business parties use this class to access the registry
+ */
+public class RegistryCenter {
+
+    private static final Logger logger = LoggerFactory.getLogger(RegistryCenter.class);
+
+    private final AtomicBoolean isStarted = new AtomicBoolean(false);
+
+    private Registry registry;
+
+    private IStoppable stoppable;
+
+    /**
+     * nodes namespace
+     */
+    protected static String NODES;
+
+    /**
+     * master path
+     */
+    protected static String MASTER_PATH = "/nodes/master";
+
+    private RegistryPluginManager registryPluginManager;
+    /**
+     * worker path
+     */
+    protected static String WORKER_PATH = "/nodes/worker";
+
+    protected static final String EMPTY = "";
+
+    private static final String REGISTRY_PREFIX = "registry";
+
+    private static final String REGISTRY_PLUGIN_BINDING = "registry.plugin.binding";
+
+    private static final String REGISTRY_PLUGIN_DIR = "registry.plugin.dir";
+
+    private static final String MAVEN_LOCAL_REPOSITORY = "maven.local.repository";
+
+    private static final String REGISTRY_PLUGIN_NAME = "registry.plugin.name";
+
+    /**
+     * default registry plugin dir
+     */
+    private static final String REGISTRY_PLUGIN_PATH = "lib/plugin/registry";
+
+    /**
+     * init node persist
+     */
+    public void init() {
```

Comment:
Currently, this init method is executed in `MasterRegistryClient` and `WorkerRegistryClient`.
And our api module will not scan the `server` module, so in `api` module, the register plugin will not be loaded in `api` module.

Reply:
This is handed over to alert for initialization. Currently, this PR is still in WIP status. When finished, I will modify the status.

Reply:
Some of the information obtained in the api belongs to the unique information of zk. I am not sure whether there is a good abstraction method. This is related to whether the specific plug-in is provided.

Reply:
> Some of the information obtained in the api belongs to the unique information of zk. I am not sure whether there is a good abstraction method. This is related to whether the specific plug-in is provided.

Yes, at present, we query two kinds of data from zookeeper in api.
One is the information of Master and Worker, these data can be provided by all kinds of Registry.
The other is zookeeper's own information, these data can be considered as registry's own metrics.